qt_ui/windows/basemenu/QBaseMenu2.py

```diff
@@ -241,10 +242,26 @@ def closeEvent(self, close_event: QCloseEvent):
         GameUpdateSignal.get_instance().updateGame(self.game_model.game)
 
     def get_base_image(self):
-        if self.cp.cptype == ControlPointType.AIRCRAFT_CARRIER_GROUP:
-            return "./resources/ui/carrier.png"
-        elif self.cp.cptype == ControlPointType.LHA_GROUP:
-            return "./resources/ui/lha.png"
+        game = self.game_model.game
+
+        if (
+            self.cp.cptype == ControlPointType.AIRCRAFT_CARRIER_GROUP
+            or self.cp.cptype == ControlPointType.LHA_GROUP
+        ):
+            carrier_type = self.cp.get_carrier_group_type()
+            if (
+                db.ship_type_from_name(carrier_type)
+                in [
+                    Stennis,
+                    KUZNECOW,
+                ]
+                and game.settings.supercarrier
+            ):
+                supercarrier_type = db.upgrade_to_supercarrier(
+                    db.ship_type_from_name(carrier_type), self.cp.name
+                ).id
```

Comment:
Any reason to not handle this within `get_carrier_group_type`? We don't want the upgrade code copy/pasted everywhere, so probably better to do that?

Reply:
I suppose not. I'll make the change.

Reply:
Upon further investigation, ControlPoint doesn't have access to game. Seems like a big refactor to add that, since the control points are generated when Campaign is created and Game doesn't exist at that point yet.